src/bindings/python/flux/job/output.py

```diff
@@ -0,0 +1,957 @@
+            jobid,
+            labelio=self.labelio,
+            log_stderr_level=self.log_stderr_level,
+            nowait=self.nowait,
+        ).then(self._output_watch_cb)
+
+    def _output_watch_cb(self, future):
+        #  Split data into multiple lines and fulfill future once per line
+        try:
+            stream, lines = future.get_output()
+        except OSError as exc:
+            self.fulfill_error(exc.errno, exc.strerror)
+            return
+        if lines is not None:
+            for line in lines.splitlines(keepends=self.keepends):
+                self.fulfill((stream, line))
+        else:
+            self.fulfill((None, None))
+
+    def getline(self):
+        """
+        Return a tuple of (stream, line) for the next line of available
+        output from a JobOutputWatch Future. Possible values for stream
+        include: "stdout", "stderr", or "log."
+
+        When no more output is available, (None, None) will be returned.
+        """
+        result = self.get()
+        self.reset()
+        return result
+
+
+def output_event_watch_async(flux_handle, jobid, labelio=False, nowait=False):
+    """
+    Asynchronously get output event updates for a job.
+
+    Returns a JobOutputEventWatch Future. Call .get_event() from the callback
+    to get the next available output event from the Future. The event will
+    be of type OutputEvent, LogEvent, OutputHeaderEvent, RedirectEvent, or
+    JobExceptionEvent (check 'name' attribute or type to get type of event)
+
+    If the job has a fatal exception before the 'start' event, then the
+    future will be fulfilled with an error with errno=EIO and message
+    'job {jobid} never started'.
+
+    Args:
+        jobid (int, JobID, str): jobid to watch
+        labelio (bool): label lines of output with source tasks (default: False)
+        nowait (bool): Assume output eventlog already exists and skip watching
+            precursor eventlogs.
+    Returns:
+        JobOutputEventWatch object
+    """
+    return JobOutputEventWatch(flux_handle, jobid, labelio=labelio, nowait=nowait)
+
+
+def output_watch_async(
+    flux_handle,
+    jobid,
+    labelio=False,
+    log_stderr_level=LOG_TRACE,
+    nowait=False,
+):
+    """
+    Asynchronously get output data for a job.
+
+    This function returns a JobOutputWatch Future. Use future.get_output()
+    to get the available output, which returns a (stream, data) tuple,
+    where stream is one of 'stdout', 'stderr', or 'log'.
+
+    If the job receives an exception while watching output, an appropriate
+    error message is emitted to the stderr stream. Similarly, redirect
+    events generate a "redirected to" message on stderr.
+
+    If the job has a fatal exception before the 'start' event, then the
+    future will be fulfilled with an error with errno=EIO and message
+    'job {jobid} never started'.
+
+    Args:
+        labelio (bool): label lines of output with source tasks (default=False)
```

Comment:
flux handle and jobid arg?  atleast jobid since you did that earlier.  also for many of the below.